gateway/src/shard/impl.rs

```diff
@@ -92,6 +99,10 @@ impl Error for CommandError {
 #[derive(Debug)]
 #[non_exhaustive]
 pub enum CommandErrorType {
+    /// Heartbeat has not been initialized yet.
+    HeartbeatNotActive,
+    /// Requesting the ratelimiter has failed because the actor has been stopped.
+    RatelimiterFailed,
```

Comment:
Failed is a little repetitive. We probably have some other enums with variants like it. Could we clarify it like `RatelimiterStopped`? Why would it stop in the first place, anyway? Is this a detail users should know, or can we prevent it somehow? Or guarantee it won't stop? Presumably if the actor has stopped then the executor has been shut down, so maybe that would be a better message?

Reply:
Since the actor never stops unless the runtime executor shuts down, I went with that approach